### PR TITLE
:rocket: feat: Add custom http proxy for http client in proxy middleware

### PR DESCRIPTION
**Pull request by li-jin-gou**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. 
Explain the *details* for making this change. What existing problem does the pull request solve?

Fixes  https://github.com/gofiber/fiber/pull/1480

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] For new functionalities I follow the inspiration of the express js framework and built them similar in usage
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Commit formatting:

Use emojis on commit messages so it provides an easy way of identifying the purpose or intention of a commit. Check out the emoji cheatsheet here: https://gitmoji.carloscuesta.me/

### CODE REVIEW

**Comment by li-jin-gou** [comment]
how to merge pull request ,I  am not familiar with this :fearful:

File: middleware/proxy/README.md [hunk 1]
```diff
@@ -121,7 +128,10 @@ type Config struct {
 	WriteBufferSize int
 
 	// tls config for the http client
-	TlsConfig *tls.Config
+	TlsConfig *tls.Config 
+	
+	// Proxy specifies an http proxy to dial http client requests through 
+	Proxy string
```
Comment (efectn) [inline middleware/proxy/README.md:134]: Please use this on Balancer and new test case for this.

Comment (li-jin-gou) [reply middleware/proxy/README.md:134]: I add a case  `Test_Proxy_Balancer_WithHttpProxy`

Comment (li-jin-gou) [reply middleware/proxy/README.md:134]: thanks

**Review by efectn — APPROVED** [review]

**Comment by li-jin-gou** [comment]
This is achieved by customizing the client and refer to https://github.com/gofiber/fiber/pull/2117.

**Pull request closed without merging**